sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionAsyncContainer.java

```diff
@@ -999,49 +1180,66 @@ public Mono<CosmosBatchResponse> executeCosmosBatch(CosmosBatch cosmosBatch) {
      * transactional batch succeeded.
      */
     public Mono<CosmosBatchResponse> executeCosmosBatch(CosmosBatch cosmosBatch, CosmosBatchRequestOptions requestOptions) {
-        if (requestOptions == null) {
-            requestOptions = new CosmosBatchRequestOptions();
-        }
+        final CosmosBatchRequestOptions cosmosBatchRequestOptions = Optional.ofNullable(requestOptions)
+            .orElse(new CosmosBatchRequestOptions());
 
         List<Mono<ItemBatchOperation<?>>> monoList = new ArrayList<>();
         for (ItemBatchOperation<?> itemBatchOperation : cosmosBatchAccessor.getOperationsInternal(cosmosBatch)) {
             Mono<ItemBatchOperation<?>> itemBatchOperationMono = null;
             if (itemBatchOperation.getItem() != null) {
-                ObjectNode objectNode =
-                    EncryptionUtils.getSimpleObjectMapper().valueToTree(itemBatchOperation.getItem());
-                itemBatchOperationMono =
-                    encryptionProcessor.encryptObjectNode(objectNode).map(encryptedItem -> {
-                        return new ItemBatchOperation<>(
+                itemBatchOperationMono = this.encryptionProcessor.initEncryptionSettingsIfNotInitializedAsync()
+                    .thenReturn(this.encryptionProcessor.getEncryptionSettings())
+                    .flatMap(encryptionSettings -> {
+                        try {
+                            Field id = itemBatchOperation.getItem().getClass().getDeclaredField(Constants.PROPERTY_NAME_ID);
```

Comment:
why are we doing this?

Reply:
In the Cosmos Bulk and batch operation, for the create and upsert operations the itemPojo is not null, but the itemId is null when we create the ItemBulkOperation/ItemBatchOperation. I am using that piece of code to get the value for the itemId from the itemPojo. Had a discussion offline as well.